app/src/main/java/uk/ac/ebi/atlas/experimentpage/ExpressionAtlasContentService.java

```diff
@@ -2,82 +2,88 @@
 
     public ExpressionAtlasContentService(
-            LinkToArrayExpress.SingleCell singleCellLinkToArrayExpress
-            ,
-            LinkToEna.SingleCell singleCellLinkToEna,
-            LinkToEga.SingleCell singleCellLinkToEga,
-            LinkToGeo.SingleCell singleCellLinkToGeo,
+            LinkToArrayExpress linkToArrayExpress,
+            LinkToEna linkToEna,
+            LinkToEga linkToEga,
+            LinkToGeo linkToGeo,
             ExperimentTrader experimentTrader) {
         this.experimentTrader = experimentTrader;
-
-        this.singleCellBaselineExperimentExternallyAvailableContentService=
-                new ExternallyAvailableContentService<>(ImmutableList.of(singleCellLinkToArrayExpress));
-
-        this.singleCellBaselineExperimentExternallyAvailableContentServiceGeo =
-                new ExternallyAvailableContentService<>(ImmutableList.of(singleCellLinkToGeo));
-
-        this.singleCellBaselineExperimentExternallyAvailableContentServiceEna =
-                new ExternallyAvailableContentService<>(ImmutableList.of(singleCellLinkToEna));
-
-        this.singleCellBaselineExperimentExternallyAvailableContentServiceEga =
-                new ExternallyAvailableContentService<>(ImmutableList.of(singleCellLinkToEga));
+        this.linkToEna = linkToEna;
+        this.linkToEga = linkToEga;
+        this.linkToGeo = linkToGeo;
+        this.linkToArrayExpress = linkToArrayExpress;
     }
 
     public List<ExternallyAvailableContent> getExternalResourceLinks(String experimentAccession,
                                                  String accessKey,
                                                  ExternallyAvailableContent.ContentType contentType) {
         Experiment<?> experiment = experimentTrader.getExperiment(experimentAccession, accessKey);
-        var externalResourceType = getExternalResouceType(experiment);
-
-        var arrayExpressResourceLinks = singleCellBaselineExperimentExternallyAvailableContentService.list(
-                (SingleCellBaselineExperiment) experiment, contentType);
-        var otherExternalResourceLinks =  externalResourceType.equals("geo") ?
-                singleCellBaselineExperimentExternallyAvailableContentServiceGeo.list(
-                (SingleCellBaselineExperiment) experiment, contentType) :
-                externalResourceType.equals("ega") ?
-                        singleCellBaselineExperimentExternallyAvailableContentServiceEga.list(
-                        (SingleCellBaselineExperiment) experiment, contentType) :
-                        singleCellBaselineExperimentExternallyAvailableContentServiceEna.list(
-                        (SingleCellBaselineExperiment) experiment, contentType);
-        otherExternalResourceLinks.addAll(arrayExpressResourceLinks);
-        return otherExternalResourceLinks;
+
+        ImmutableList.Builder<ExternallyAvailableContent> arrayExpressResourceLinks = ImmutableList.builder();
```

Comment:
I would rename it to `externalResourceLinks` as it might contains links not just to `ArrayExpress` resources.
```suggestion
        var externalResourceLinks = ImmutableList.builder();
```